controllers/utils/affinity_group.go

```diff
@@ -88,12 +89,19 @@ func (r *ReconciliationRunner) GetOrCreateAffinityGroup(
 }
 
 // The computed affinity group name relevant to this machine.
-func GenerateAffinityGroupName(csm infrav1.CloudStackMachine, capiMachine *clusterv1.Machine) (string, error) {
+func GenerateAffinityGroupName(csm infrav1.CloudStackMachine, capiMachine *clusterv1.Machine, capiCluster *clusterv1.Cluster) (string, error) {
 	managerOwnerRef := GetManagementOwnerRef(capiMachine)
 	if managerOwnerRef == nil {
 		return "", errors.Errorf("could not find owner UID for %s/%s", csm.Namespace, csm.Name)
 	}
 	titleCaser := cases.Title(language.English)
-	return fmt.Sprintf("%sAffinity-%s-%s-%s",
-		titleCaser.String(csm.Spec.Affinity), managerOwnerRef.Name, managerOwnerRef.UID, csm.Spec.FailureDomainName), nil
+
+	// If the machine's owner is KubeadmControlPlane or EtcdadmCluster, then we don't consider the name and UID of the
+	// owner, since there will only be one of each of those per cluster.
+	if managerOwnerRef.Kind == "KubeadmControlPlane" || managerOwnerRef.Kind == "EtcdadmCluster" {
```

Comment:
@vishesh92 
there could be multiple control plane nodes in the cluster, right ?
cc @vignesh-goutham

Reply:
As per my understanding, control nodes for a capi cluster & a failure domain should have the same config. And we are using cluster's name & uuid to ensure separate node group for each capi cluster.